icu4c/source/i18n/formattedvalue.cpp

```diff
@@ -193,6 +193,12 @@ ucfpos_close(UConstrainedFieldPosition* ptr) {
 }
 
 
+// -Wreturn-local-addr first found in https://gcc.gnu.org/onlinedocs/gcc-4.8.5/gcc/Warning-Options.html#Warning-Options
+#if U_GCC_MAJOR_MINOR >= 409
+#pragma GCC diagnostic push
+#pragma GCC diagnostic warning "-Wreturn-local-addr"
```

Comment:
This line doesn't serve any purpose, does it? Then it should be deleted.

Reply:
I can remove it. I think I hit an issue that some version of the compiler won't support the ignore and I have to add this line. But I will try to remove it to see will it still cause problem.